garden-service/src/garden.ts

```
@@ -294,6 +299,13 @@ export class Garden {
         throw new InternalError(errMsg, {})
       } else {
         await login(platformUrl, log)
+        secrets = await getSecrets({
```

What happens if `secrets` don't contain all the secrets the config expects. Say you have something like:

```yaml
services:
# ...
  env:
    API_KEY: ${secrets.apiKey}
```

and `secrets.apiKey` is undefined. 

Does it just deploy the service with an empty `API_KEY`? Or is this handled somewhere?

---

I think this is a valid question, from a fail early point of view, I think garden should fail that secrets were not resolved. I think that is what is currently happening

---

An error will be thrown if the key is not defined (see https://github.com/garden-io/garden/blob/secrets-in-template-strings/garden-service/test/unit/src/config/config-context.ts#L343).

This leans on our generic error handling for undefined keys in template strings.

We are not, however, validating that the value is truthy (so a secret with a `null`/`undefined` value for a valid/present key would not result in an error).

---

All right. Wondering how we should in general handle platforrm related errors. For example if the secret is missing from the backend you want a better error than just: "Failed resolving template string..." (or whatever it is).

---

I've added some error handling around empty secret values (in a TBS commit).